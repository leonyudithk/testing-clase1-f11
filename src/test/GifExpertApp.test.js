import React from 'react'
import {shallow} from 'enzyme'
import {GifExpertApp} from '../GifExpertApp'

describe('Pruebas al Componente GifExpertApp', ()=>{
    test('Debe mostrarse correctamente una categoria', ()=>{
            //Arreglar-preparar mi mesa de trabajo
            const wrapper = shallow(<GifExpertApp/>)

            //Actuar que es hacer el test
            expect(wrapper).toMatchSnapshot()
    })

    test('Mostrar una lista categoria', ()=>{
        //Arreglar-preparar mi mesa de trabajo
        const categorias = ['Marvel', 'Flores', 'Holas', 'Dragon Ball', '#']
        const wrapper = shallow(<GifExpertApp  defaultCategories={categorias} />)

        //Actuar que es hacer el test
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categorias.length)

   
})
})