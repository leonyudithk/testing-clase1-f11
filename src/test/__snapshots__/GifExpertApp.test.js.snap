// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al Componente GifExpertApp Debe mostrarse correctamente una categoria 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas al Componente GifExpertApp Mostrar una lista categoria 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
    <GifGrid
      category="Flores"
      key="Flores"
    />
    <GifGrid
      category="Holas"
      key="Holas"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="#"
      key="#"
    />
  </ol>
</Fragment>
`;
